File: src/main/java/io/tbbc/cf/common/Size.java
package io.tbbc.cf.common;

public enum Size {
    SMALL("s", "small"),
    MEDIUM("m", "medium"),
    LARGE("l", "large"),
    EXTRA_LARGE("xl", "extra large");

    private final String egoInitial;
    private final String egoFullString;

    Size(String egoInitial, String egoFullString) {
        this.egoInitial = egoInitial;
        this.egoFullString = egoFullString;
    }

    public String egoInitial() {
        return egoInitial;
    }

    public String egoFullString() {
        return egoFullString;
    }
}

File: src/main/webui/src/model/common/Size.ts
export default class Size {
    key:string;
    label:string;
    private constructor(key:string, label:string) {
        this.key = key;
        this.label = label;
    }
    static SMALL: Size = new Size("SMALL", "Small");
    static MEDIUM: Size = new Size("MEDIUM", "Medium");
    static LARGE: Size = new Size("LARGE", "Large");
    static EXTRA_LARGE: Size = new Size("EXTRA_LARGE", "Extra Large");
    static values() : Size[] {
        return [Size.SMALL, Size.MEDIUM, Size.LARGE, Size.EXTRA_LARGE];
    }
    static fromKey(key: string) : Size {
        return Size.values().find((size) => size.key === key);
    }
}